server.go

```diff
@@ -355,12 +378,38 @@ func (s *Server) StartAsSubserver(lndClient lnrpc.LightningClient,
 		}
 	}()
 
-	// Start the macaroon service and let it create its default macaroon in
-	// case it doesn't exist yet.
-	if err := s.startMacaroonService(createDefaultMacaroonFile); err != nil {
-		return err
+	if createDefaultMacaroonFile {
```

Comment:
macaroon db not used if in subserver + stateless mode

Reply:
Maybe it would be better to rename this to `withOwnMacaroonValidation` or `withMacaroonService` since this variable does a bit more now?